### PR TITLE
[tune/internal] Move signal handling into separate method

### PR DESCRIPTION
**Pull request by krfricke**
Signed-off-by: Kai Fricke <kai@anyscale.com>

## Why are these changes needed?

This PR improves the tune.py code structure:

- Signal handling is set up in a separate method
- Signal handling now uses a threading event instead of a global dict
- The GPU check is moved to before a trial runner is instantiated (and thus restored) in order to fail faster if fail conditions are met
- Trial executor creation is moved closer to where the trial executor is passed

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/tune.py [hunk 1]
```diff
@@ -127,6 +127,39 @@ def _report_progress(
         reporter.report(trials, done, sched_debug_str, executor_debug_str)
 
 
+def _setup_signal_catching() -> threading.Event:
+    original_handler = signal.getsignal(signal.SIGINT)
+    stop_event = threading.Event()
```
Comment (xwjiang2010) [inline python/ray/tune/tune.py:132]: more specific name? `tune_loop_interrupted_event` or something alike?

**Review by xwjiang2010 — APPROVED** [review]
Nice! Thanks!

File: python/ray/tune/tune.py (hunk 1 above)
Comment (krfricke) [reply python/ray/tune/tune.py:132]: If you're ok with it I'd like to keep this, as we set up general signal catching here and not just the event. Let me know if you disagree!
Meanwhile I'll kick of tune cloud release tests

Comment (xwjiang2010) [reply python/ray/tune/tune.py:132]: oh I mean the event name can be more specific? The method name `_setup_signal_catching` sounds fine to me!

Comment (krfricke) [reply python/ray/tune/tune.py:132]: Ah I see! Yes, thanks, will updated!

**Review by Yard1 — APPROVED** [review]

**Pull request merged**